py/torch_tensorrt/dynamo/compile.py

```diff
@@ -0,0 +1,183 @@
+from torch_tensorrt.dynamo.backend import torch_tensorrt_backend
+from torch_tensorrt.dynamo.conversion import convert_module
+
+from torch_tensorrt.dynamo._defaults import (
+    PRECISION,
+    DEBUG,
+    WORKSPACE_SIZE,
+    MIN_BLOCK_SIZE,
+    PASS_THROUGH_BUILD_FAILURES,
+    MAX_AUX_STREAMS,
+    VERSION_COMPATIBLE,
+    OPTIMIZATION_LEVEL,
+    USE_PYTHON_RUNTIME,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+def compile(
+    gm: torch.nn.Module,
+    inputs: Any,
+    *,
+    device=Device._current_device(),
+    disable_tf32=False,
+    sparse_weights=False,
+    enabled_precisions=set(),
+    refit=False,
+    debug=DEBUG,
+    capability=EngineCapability.default,
+    num_avg_timing_iters=1,
+    workspace_size=WORKSPACE_SIZE,
+    dla_sram_size=1048576,
+    dla_local_dram_size=1073741824,
+    dla_global_dram_size=536870912,
+    calibrator=None,
+    truncate_long_and_double=False,
+    require_full_compilation=False,
+    min_block_size=MIN_BLOCK_SIZE,
+    torch_executed_ops=[],
+    torch_executed_modules=[],
+    pass_through_build_failures=PASS_THROUGH_BUILD_FAILURES,
+    max_aux_streams=MAX_AUX_STREAMS,
+    version_compatible=VERSION_COMPATIBLE,
+    optimization_level=OPTIMIZATION_LEVEL,
+    use_python_runtime=USE_PYTHON_RUNTIME,
+    **kwargs,
+):
+    if debug:
+        logger.setLevel(logging.DEBUG)
+
+    logger.warn(
+        "The Dynamo backend is an experimental feature, for which only the "
+        + "following arguments are supported: "
+        + "{enabled_precisions, debug, workspace_size, min_block_size, "
+        + "torch_executed_ops, pass_through_build_failures}"
+    )
+
+    if not isinstance(inputs, collections.abc.Sequence):
+        inputs = [inputs]
+
+    torchtrt_inputs, torch_inputs = prepare_inputs(inputs, prepare_device(device))
+
+    if (
+        torch.float16 in enabled_precisions
+        or torch_tensorrt.dtype.half in enabled_precisions
+    ):
+        precision = torch.float16
+    elif (
+        torch.float32 in enabled_precisions
+        or torch_tensorrt.dtype.float in enabled_precisions
+    ):
+        precision = torch.float32
+    elif len(enabled_precisions) == 0:
+        logger.info(f"No precision specified, defaulting to {PRECISION}")
+        precision = PRECISION
+    else:
+        raise ValueError(
+            f"Precision {enabled_precisions} not supported in the Dynamo Path"
+        )
+
+    compilation_options = {
+        "precision": precision,
+        "debug": debug,
+        "workspace_size": workspace_size,
+        "min_block_size": min_block_size,
+        "torch_executed_ops": torch_executed_ops,
+        "pass_through_build_failures": pass_through_build_failures,
+        "max_aux_streams": max_aux_streams,
+        "version_compatible": version_compatible,
+        "optimization_level": optimization_level,
+        "use_python_runtime": use_python_runtime,
+    }
+
+    if kwargs.get("ir", "dynamo") == "torch_compile":
```

Comment:
torch_tensorrt.dynamo.compile should be only for export

Reply:
done